.github/workflows/scripts/bump-versions.sh

```diff
@@ -0,0 +1,210 @@
+#!/bin/bash
+
+# This script takes either current beta version, current stable version, or both,
+# and optionally next beta version and next stable version, and updates the version
+# in the specified files. If next version is not provided, it will infer the next 
+# semantic version (e.g. v0.110.0 -> v0.111.0 or v1.16.0 -> v1.17.0) based on the 
+# current version(s) passed.
+
+# Function to display usage
+usage() {
+  echo "Usage: $0 [--commit] [--pull-request] --current-beta-core <current-beta-core> [--current-beta-contrib <current-beta-contrib>] [--current-stable <current-stable>] [--next-beta-core <next-beta-core>] [--next-beta-contrib <next-beta-contrib>] [--next-stable <next-stable>]"
+  echo "  --current-beta-core: Current beta version of the core component (e.g., v0.110.0)"
+  echo "  --current-beta-contrib: Current beta version of the contrib component (e.g., v0.110.0)"
+  echo "  --current-stable: Current stable version of the core component (e.g., v1.16.0)"
+  echo "  at least one of the above three arguments is required, or there is no version to update."
+  echo
+  echo "  --next-beta-core: Next beta version of the core component (e.g., v0.111.0)"
+  echo "  --next-beta-contrib: Next beta version of the contrib component (e.g., v0.111.0)"
+  echo "  --next-stable: Next stable version of the core component (e.g., v1.17.0)"
+  echo
+  echo "  --commit: Commit the changes to a new branch"
+  echo "  --pull-request: Push the changes to the repo and create a draft PR (requires --commit)"
+  exit 1
+}
+
+# Function to validate semantic version and strip leading 'v'
+validate_and_strip_version() {
+  local var_name=$1
+  local version=${!var_name}
+  # Strip leading 'v' if present
+  version=${version#v}
+  if [[ ! $version =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    echo "Invalid version: $version. Must be a semantic version (e.g., 1.2.3)."
+    exit 1
+  fi
+  eval "$var_name='$version'"
+}
+commit_changes=false
+create_pr=false
+# Parse named arguments
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    --current-beta-core) current_beta_core="$2"; shift ;;
```

Comment:
Thinking about this more, do we actually even need the "current" versions from the user? Seems that we can read the current versions from the manifest files, e.g. with `awk`:

```console
$ current_beta_core=$(awk '/^.*go\.opentelemetry\.io\/collector\/.* v0/ {print $4; exit}' distributions/otelcol/manifest.yaml)
$ echo $current_beta_core
v0.111.0
```